app/controllers/contracts_controller.rb

```diff
@@ -0,0 +1,46 @@
+class ContractsController < ApplicationController
+  protect_from_forgery with: :null_session
+
+  def index
+    contracts = []
+    Contract.all.each do |contract|
+      contracts.push(format_contract(contract))
+    end
+    render json: contracts
+  end
+
+  def show
+    contract = Contract.find(params[:id])
+    render json: format_contract(contract)
+  end
+
+  def update
+    contract = Contract.find(params[:id])
+    contract.update_attributes!(contract_params)
+  end
+
+  def nag
+    contract = Contract.find(params[:id])
+    # send out email remainders
+    contract.update_attributes!(nag_count: contract[:nag_count]+1)
```

Comment:
We could use [`increment`/`increment!`](http://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-increment) here.